sdk/communication/azure-communication-rooms/src/main/java/com/azure/communication/rooms/RoomsAsyncClient.java

```diff
@@ -626,4 +614,11 @@ private Map<String, ParticipantProperties> convertRoomIdentifiersToMapForRemove(
         return participantMap;
     }
 
+    private static String getUpdateRequest(Map<String, ParticipantProperties> participantMap) throws IOException {
+        Writer json = new StringWriter();
+        try (JsonWriter jsonWriter = JsonProviders.createWriter(json)) {
+            new UpdateParticipantsRequest().setParticipants(participantMap).toJson(jsonWriter);
```

Comment:
This is missing a call to `flush()`, right now this content written to the writer may not be complete without flushing if there is buffering happening within the JsonWriter itself.

Reply:
Forgot that JsonWriter flushes on close